duffel_api/models/airport.py

```diff
@@ -1,21 +1,36 @@
-class Airport:
-    """Airports are used to identify origins and destinations in journey slices"""
+from dataclasses import dataclass
+from typing import Optional
+
+from ..models import City
+from ..utils import get_and_transform
 
-    def __init__(self, json):
-        for key in json:
-            value = json[key]
-            if key == "city" and value:
-                setattr(self, key, City(value))
-            else:
-                setattr(self, key, value)
 
+@dataclass
+class Airport:
+    """Airports are used to identify origins and destinations in journey
+    slices"""
 
-class City:
-    """The metropolitan area where the airport is located.
-    Only present for airports which are registered with IATA as
-    belonging to a metropolitan area.
-    """
+    id: str
+    name: str
+    iata_code: Optional[str]
+    icao_code: Optional[str]
+    country_code: str
+    latitude: float
+    longitude: float
+    time_zone: str
+    city: Optional[City]
 
-    def __init__(self, json):
-        for key in json:
-            setattr(self, key, json[key])
+    @classmethod
+    def from_json(cls, json: dict):
+        """Construct a class instance from a JSON response."""
+        return cls(
+            id=json["id"],
+            name=json["name"],
+            iata_code=json.get("iata_code"),
+            icao_code=json.get("icao_code"),
+            country_code=json["country_code"],
+            latitude=json["latitude"],
+            longitude=json["longitude"],
+            time_zone=json["time_zone"],
+            city=get_and_transform(json, "city", City.from_json),
```

Comment:
Another one-liner option is:

```python
            city=City.from_json(json["city"]) if "city" in json else None,
```

Or even handle the `None` case inside `City.from_json` maybe?

Reply:
Resolving as being discussed elsewhere.